### PR TITLE
KAS-4099 implement removing sign-flow data from signature tab

### PR DESCRIPTION
**Pull request by ValenberghsSven**
https://kanselarij.atlassian.net/browse/KAS-4099
https://kanselarij.atlassian.net/browse/KAS-2793

Original PR: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1751
This branch contains fixes to the signature-service method to completely remove all data models.
- Added a new permission to remove signflow related data
- Added a service call to clean up everything related to a sign flow.
- Added dropdown and action in document details panel in doc viewer (with permission)
- Added verify modal for this destructive action in document details panel 
- Added action with tooltip in signature tab
- Added verify modal for this destructive action in signature tab

Added serializer because we set some `belongsTo` relations inversely (activity knows sign-subcase)
Some data might be added by the digital-signing service.
Saving signSubcase after loading all relations may result in overwriting lists with stale data (no reload before save).

Also added the `hasMany` to serialize.
Even though Ember should not serialize it (saving sign-subcase does not save the list (This used to happen in earlier ember versions)) > deleting sign-subcase after deleting all hasMany activities yields an error in console.
image

This error is not present when reloading the `signsubcase` model before deleting.

### CODE REVIEW

**Comment by ValenberghsSven** [comment]
The modals are staying up during the process, allowing double clicks.
Need to prevent those...

File: app/components/web-components/vl-modal-verify.hbs [hunk 1]
```diff
@@ -6,7 +6,7 @@
   {{! template-lint-disable no-down-event-binding}}
   {{on "keydown" this.keyDown}}
 >
-  <div class="auk-modal auk-modal--small" role="dialog">
+  <div class="auk-modal auk-modal--medium" role="dialog">
```
Comment (ValenberghsSven) [inline app/components/web-components/vl-modal-verify.hbs:9]: We only use this modal in 3 places (coming from one). Small is not big enough.

File: app/components/documents/document-details-panel.hbs [hunk 2]
```diff
@@ -179,23 +179,39 @@
...
+                @iconAlignment="left"
+                download
+              >
+                {{@piece.signedPiece.name}}.{{@piece.signedPiece.file.extension}}
+              </AuLinkExternal>
+              <AccessLevelPill 
+                @accessLevel={{await @piece.signedPiece.accessLevel}} 
+                @isEditable={{false}}
+              />
+              <AuDropdown
```
Comment (tdn) [inline app/components/documents/document-details-panel.hbs]: I think a `{{#if (user-may "remove-signatures")}}` would be appropriate here. Now this is only checked by `this.currentSession.may('manage-signatures');` in `isSignaturesEnabled`.

If we have a separate permission for removal, might as well use it consistently.

File: app/components/documents/document-card.hbs [hunk 3]
```diff
@@ -259,4 +259,4 @@
   @confirmMessage={{t "delete"}}
   @confirmIcon="trash"
   @alert={{true}}
-/>
+/>
```
Comment (tdn) [inline app/components/documents/document-card.hbs:262]: Just a sidenote, as this wasn't specified in the original issue. When we delete a document container now without first deleting the signed piece, the signed document still lives in the db and on the system (it's now orphaned).

We have a follow-up issue for that (KAS-2794), but I'm wondering if we should disable the removal of pieces that have signed pieces for now. That would force the user to first delete the signed piece, and then the original.

**Review by tdn — CHANGES_REQUESTED** [review]
Two small points:
1. Use `may remove-signatures` for the action next to the pill on the 'Handtekeningen' tab
2. Disable the container removal action for original pieces with a signed piece to avoid keeping orphaned signed pieces in the db.

File: app/components/documents/document-card.hbs (hunk 3 above)
Comment (ValenberghsSven) [reply app/components/documents/document-card.hbs:262]: This is not the only place where you can remove document containers. (batch edit will also delete&cleanup / document-card actions..) 
I would argue we fix this separately

**Comment by ValenberghsSven** [comment]
> Two small points:
> 
>     1. Use `may remove-signatures` for the action next to the pill on the 'Handtekeningen' tab
> 
>     2. Disable the container removal action for original pieces with a signed piece to avoid keeping orphaned signed pieces in the db.

1. I think you meant document-details-panel
2. out of scope? this is not the only place where we can delete pieces/containers. Best to dedicate it's own ticket.

**Comment by ValenberghsSven** [comment]
I'm also of the opinion that the use-case where a user would delete an entire document after it has been signed would be user error or emergency action. Should not happen to often.
Do we want to disable this in total? Admin should be able to do it.
Maybe it is better to include the cleanup in the deletion process regardless and add additional warnings/information in the verify popup?

**Review by tdn — APPROVED** [review]

**Pull request merged**